shared/src/lib.rs

```diff
@@ -83,3 +84,12 @@ pub fn debug_bytes(
 pub fn clone_anyhow_error(err: &anyhow::Error) -> anyhow::Error {
     anyhow::anyhow!("{:#}", err)
 }
+
+#[derive(Eq, Hash, PartialEq)]
+pub struct AppId(pub [u8; 32]);
+
+impl Debug for AppId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
```

Comment:
Just so you know, the `<'_>` can li4kely be omitted here:
```suggestion
    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
```